src/auth/index.js

```diff
@@ -160,14 +161,18 @@ export default class Auth {
 
     document.cookie = `${authCommon.USER_AUTH_COOKIE_NAME}=; Path=/; Expires=Thu, 01 Jan 1970 00:00:00 GMT;`;
     const userAuth = this.unmarshallUserAuth(uaCookie);
-    return this.set(userAuth).then(() =>
+    return this.getLoggedInProviderType().then(providerType => this.set(userAuth, providerType)).then(() =>
```

Comment:
[major] How does the ordering work here? `getLoggedInProviderType` wouldn't be able to return anything useful when it's the first time we're here and as a result, `set` will fail due to the restriction it has in place about empty types.

Reply:
This one isn't like redirect and in this special case, we can assume the providerType is `mongodb-cloud`.